streams/src/test/java/org/apache/kafka/test/StreamsTestUtils.java

```diff
@@ -273,4 +281,70 @@ public static boolean isCheckSupplierCall() {
         return Arrays.stream(Thread.currentThread().getStackTrace())
                 .anyMatch(caller -> "org.apache.kafka.streams.internals.ApiUtils".equals(caller.getClassName()) && "checkSupplier".equals(caller.getMethodName()));
     }
+
+    public static StreamTask createStatefulTask(final TaskId taskId,
```

Comment:
nit: we have other unit test classes duplicating this logic, e.g. in the `DefaultStateUpdaterTest` above, also in `StreamTaskTest#createStatelessTask`. Could we consolidate them all in this class?

Reply:
That was the idea. I just did not want to pollute too much this PR.

Reply:
BTW, you do not want to use a mock task in `StreamTaskTest` since you are testing a task there.

Reply:
Ack.